### PR TITLE
[Doc] Add hive meta cache to hive catalog

### PR DESCRIPTION
**Pull request by amber-create**

### CODE REVIEW

File: data_source/catalog/hive_catalog.md [hunk 1]
```diff
@@ -430,6 +430,7 @@ StarRocks 默认采用自动异步更新策略，开箱即用。因此，一般
 | remote_file_cache_refresh_interval_sec | 否       | StarRocks 异步更新缓存的 Hive 表或分区的数据文件的元数据的时间间隔。单位：秒。默认值：`60`。 |
 | metastore_cache_ttl_sec                | 否       | StarRocks 自动淘汰缓存的 Hive 表或分区的元数据的时间间隔。单位：秒。默认值：`86400`，即 24 小时。 |
 | remote_file_cache_ttl_sec              | 否       | StarRocks 自动淘汰缓存的 Hive 表或分区的数据文件的元数据的时间间隔。单位：秒。默认值：`129600`，即 36 小时。 |
+| enable_cache_list_names                | 否       | 指定 StarRocks 是否缓存 Hive Partition Names。取值范围：`true` 和 `false`。默认值：`false`。取值为 `true` 表示开启缓存，取值为 `false` 表示关闭缓存。 |
```
Comment (stephen-shelby) [inline data_source/catalog/hive_catalog.md:433]: After 3.0, the default value is `true`

File: data_source/catalog/hive_catalog.md [hunk 2]
```diff
@@ -796,16 +797,38 @@ DROP Catalog hive_catalog_glue;
...
 
+以下情况适用于执行手动更新元数据：
+
+- 已有分区内的数据文件发生变更，如执行过 `INSERT OVERWRITE ... PARTITION ...` 命令。
+- Hive 表有 Schema 变更。
+- Hive 表被 DROP 后重建一个同名 Hive 表。
+- 创建 Hive Catalog 时在 `PROPERTIES` 中指定了 `"enable_cache_list_names" = "true"`。在 Hive 侧新增分区后，需要查询新增分区。
+  > **说明**
+  >
+  > 自 2.5.5 版本起，StarRocks 支持周期性刷新 Hive 元数据。开启 Hive 元数据周期性刷新功能以后，默认情况下 StarRocks 每 10 分钟刷新一次 Hive 元数据。因此，一般情况下，无需执行手动更新。您只有在新增分区后，需要立即查询新增分区的数据时，才需要执行手动更新。
```
Comment (stephen-shelby) [inline data_source/catalog/hive_catalog.md:828]: `周期性刷新Hive元数据缓存` may be more precise

File: data_source/catalog/hudi_catalog.md [hunk 3]
```diff
@@ -790,16 +791,25 @@ DROP Catalog hudi_catalog_glue;
...
 ### 手动更新
 
-默认情况下，StarRocks 会缓存 Hudi 的元数据、并以异步模式自动更新缓存的元数据，从而提高查询性能。此外，在对 Hudi 表做了表结构变更、或其他表更新后，您也可以使用 [REFRESH EXTERNAL TABLE](../../sql-reference/sql-statements/data-definition/REFRESH%20EXTERNAL%20TABLE.md) 更新该表的元数据，从而确保 StarRocks 第一时间生成合理的查询计划：
+默认情况下，StarRocks 会缓存 Hudi 的元数据、并以异步模式自动更新缓存的元数据，从而提高查询性能。此外，在对 Hudi 表做了表结构变更、或其他表更新后，您也可以使用 [REFRESH EXTERNAL TABLE](../../sql-reference/sql-statements/data-definition/REFRESH%20EXTERNAL%20TABLE.md) 手动更新该表的元数据，从而确保 StarRocks 第一时间生成合理的查询计划：
 
 ```SQL
 REFRESH EXTERNAL TABLE <table_name>
 ```
 
+以下情况适用于执行手动更新元数据：
```
Comment (stephen-shelby) [inline data_source/catalog/hudi_catalog.md:804]: hudi先不用写这些，后面可能会修改，这些只在hive catalog里面先放着。

File: data_source/catalog/hive_catalog.md [hunk 4]
```diff
@@ -796,16 +797,38 @@ DROP Catalog hive_catalog_glue;
...
 ```
 
-## 更新缓存元数据
+## 周期性刷新元数据
+
+自 2.5.5 版本起，StarRocks 可以周期性刷新经常访问的 Hive 外部数据目录的元数据，达到感知数据更新的效果。您可以通过以下 [FE 参数](../../administration/Configuration.md#fe-配置项)配置 Hive 元数据周期性刷新：
+
+| 配置名称                                                      | 默认值                        | 说明                                  |
+| ------------------------------------------------------------ | ---------------------------- | ------------------------------------ |
+| enable_background_refresh_connector_metadata                 | v3.0 为 true，v2.5 为 false   | 是否开启 Hive 元数据周期性刷新。开启后，StarRocks 通过轮询经常访问的 Hive 外部数据目录的 Hive Metastore 感知数据更新，并将信息缓存至对应外部数据目录的元数据中。`true` 代表开启，`false` 代表关闭。[FE 静态参数](../../administration/Configuration.md#配置-fe-静态参数)，需要在 FE 配置文件 **fe.conf** 中修改。 |
```
Comment (Youngwb) [inline data_source/catalog/hive_catalog.md:806]: This is mutable

File: data_source/catalog/hudi_catalog.md [hunk 5]
```diff
@@ -424,6 +424,7 @@ StarRocks 默认采用自动异步更新策略，开箱即用。因此，一般
 | remote_file_cache_refresh_interval_sec | 否       | StarRocks 异步更新缓存的 Hudi 表或分区的数据文件的元数据的时间间隔。单位：秒。默认值：`60`。 |
 | metastore_cache_ttl_sec                | 否       | StarRocks 自动淘汰缓存的 Hudi 表或分区的元数据的时间间隔。单位：秒。默认值：`86400`，即 24 小时。 |
 | remote_file_cache_ttl_sec              | 否       | StarRocks 自动淘汰缓存的 Hudi 表或分区的数据文件的元数据的时间间隔。单位：秒。默认值：`129600`，即 36 小时。 |
+| enable_cache_list_names                | 否       | 指定 StarRocks 是否缓存 Hive Partition Names。取值范围：`true` 和 `false`。默认值：`false`。取值为 `true` 表示开启缓存，取值为 `false` 表示关闭缓存。 |
```
Comment (Youngwb) [inline data_source/catalog/hudi_catalog.md:427]:
```suggestion
| enable_cache_list_names                | 否       | 指定 StarRocks 是否缓存 Hudi Partition Names。取值范围：`true` 和 `false`。默认值：`false`。取值为 `true` 表示开启缓存，取值为 `false` 表示关闭缓存。 |
```

File: data_source/catalog/hive_catalog.md (hunk 1 above)
Comment (amber-create) [reply data_source/catalog/hive_catalog.md:433]: A new PR will be submitted for 3.0

File: data_source/catalog/hive_catalog.md [hunk 6]
```diff
@@ -796,16 +797,28 @@ DROP Catalog hive_catalog_glue;
 INSERT INTO default_catalog.olap_db.olap_tbl SELECT * FROM hive_table
 ```
 
-## 更新缓存元数据
+## 手动或自动更新元数据
 
 ### 手动更新
 
-默认情况下，StarRocks 会缓存 Hive 的元数据、并以异步模式自动更新缓存的元数据，从而提高查询性能。此外，在对 Hive 表做了表结构变更、或其他表更新后，您也可以使用 [REFRESH EXTERNAL TABLE](../../sql-reference/sql-statements/data-definition/REFRESH%20EXTERNAL%20TABLE.md) 更新该表的元数据，从而确保 StarRocks 第一时间生成合理的查询计划：
+默认情况下，StarRocks 会缓存 Hive 的元数据、并以异步模式自动更新缓存的元数据，从而提高查询性能。此外，在对 Hive 表做了表结构变更、或其他表更新后，您也可以使用 [REFRESH EXTERNAL TABLE](../../sql-reference/sql-statements/data-definition/REFRESH%20EXTERNAL%20TABLE.md) 手动更新该表的元数据，从而确保 StarRocks 第一时间生成合理的查询计划：
```
Comment (EsoragotoSpirit) [inline data_source/catalog/hive_catalog.md:804]: image
顿号可删

**Review by EsoragotoSpirit — CHANGES_REQUESTED** [review]

**Review by EsoragotoSpirit — APPROVED** [review]

**Comment by amber-create** [comment]
@mergify backport 3.0 2.5

**Pull request merged**